packages/view/backend/sirius-components-view-emf/src/main/java/org/eclipse/sirius/components/view/emf/operations/AddExecutor.java

```diff
@@ -0,0 +1,87 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.components.view.emf.operations;
+
+import java.util.Collection;
+
+import org.eclipse.emf.ecore.EAttribute;
+import org.eclipse.emf.ecore.EDataType;
+import org.eclipse.emf.ecore.EEnum;
+import org.eclipse.emf.ecore.EEnumLiteral;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.EStructuralFeature;
+import org.eclipse.emf.ecore.util.EcoreUtil;
+import org.eclipse.sirius.components.view.emf.operations.api.IAddExecutor;
+import org.springframework.stereotype.Service;
+
+/**
+ * Used to perform the addition of a value in an object.
+ *
+ * @author sbegaudeau
+ */
+@Service
+public class AddExecutor implements IAddExecutor {
+    @Override
+    public Object eAdd(EObject instance, String featureName, Object value) {
+        var eStructuralFeature = instance.eClass().getEStructuralFeature(featureName);
+        if (eStructuralFeature == null) {
+            return null;
+        }
+
+        if (eStructuralFeature.isMany()) {
+            Object object = instance.eGet(instance.eClass().getEStructuralFeature(featureName));
```

Comment:
Why not reuse `eStructuralFeature` here?

Reply:
Because it was almost verbatim the code from Sirius Desktop :)